timemachine/cpp/src/tibd_exchange_move.cu

```diff
@@ -229,17 +228,25 @@ void TIBDExchangeMove<RealType>::move(
 
         k_attempt_exchange_move_targeted<RealType><<<ceil_divide(N, tpb), tpb, 0, stream>>>(
             N,
+            this->num_target_mols_,
             d_targeting_inner_vol_.data,
             d_box_volume_.data,
             inner_volume_,
-            this->d_uniform_noise_buffer_.data + 1, // Offset to get the last value for the acceptance criteria
+            // Offset to get the last value for the acceptance criteria
```

Comment:
would be nice to just pre-generate all these values of size NUM_MC_MOVES without dealing with offsets in opaque containers

eg. if NUM_MC_MOVES=5000

we'd have `5000*3` sized array for d_translations, and `5000*4` array for `d_quaternions`, and `5000*(NUM_WATER_MOLS)` for gumbel max, and `5000*1` for the acceptance criteria

Reply:
(we could definitely do this in a separate PR)